tagstore/tagstore_rest/v1/views/entities.py

```diff
@@ -62,5 +67,69 @@ def tags(self, request, entity_type=None, pk=None):
             queryset = entity.tags.all()
 
         extracted = [self._convert(tag) for tag in queryset]
-        serializer = EntityTagSerializer({'tags': extracted})
+        return EntityTagSerializer({'tags': extracted})
+
+    def tags(self, request, entity_type=None, pk=None):
+        '''
+        Get a list of all tags belonging to an entity.
+
+        These can be optionally filtered by `taxonomy` params.
+        '''
+        entity = get_object_or_404(Entity, external_id=pk, entity_type=entity_type)
+        taxonomies = request.GET.get('taxonomies', '').split(',')
+
+        serializer = self.serialize_tags(entity, taxonomies)
         return Response(serializer.data)
+
+    def update_tags(self, request, entity_type=None, pk=None):
+        '''
+        Update tags belonging to an entity.
+        '''
+        tagstore = DjangoTagstore()
+
+        entity_obj = get_object_or_404(Entity, external_id=pk, entity_type=entity_type)
+        entity = EntityId(external_id=pk, entity_type=entity_type)
+        freeform_obj, _ = Taxonomy.objects.get_or_create(name='FreeForm')
+        freeform = TaxonomyId(freeform_obj.id)
+
+        if not request.data.get('tags', ''):
+            return Response(status=204)
+
+        for tag in request.data['tags']:
+            # a single free form tag
+            if isinstance(tag, str):
+                try:
+                    _tag = tagstore.get_tag_in_taxonomy(tag, freeform)
+                    if not _tag:
+                        _tag = tagstore.add_tag_to_taxonomy(tag, freeform)
+                    tagstore.add_tag_to(_tag, entity)
+                except ValueError:
+                    return Response(status=500)
+
+            # a tag within a taxonomy
+            elif isinstance(tag, dict):
+                taxonomy_uid = tag.get('taxonomy_uid', None)
+                taxonomy_name = tag.get('taxonomy_name', None)
```

Comment:
I don't think we should allow selecting a taxonomy by name. Since names aren't unique, it can lead to errors.